### PR TITLE
feat: find close version fallback

### PR DESCRIPTION
**Pull request by iowillhoit**
### What does this PR do?
There are situations where we create a patch mid-week (or multiple in a single release). Sometimes the versions our customers have installed do not match what is in the Release Notes markdown. This change will look for a nearby version based on the `major.minor.x`. A warning will be shown that we are displaying RN for a nearby different version

### What issues does this PR fix or reference?
[@W-10323726@](https://gus.my.salesforce.com/apex/ADM_WorkLocator?bugorworknumber=W-10323726)

### CODE REVIEW

File: src/shared/parseReleaseNotes.ts [hunk 1]
```diff
@@ -6,37 +6,56 @@
...
-      return token;
+    if (!tokens.length) {
+      if (findClosest === false) {
+        findClosest = true;
+
+        const semverRange = `${semver.major(version)}.${semver.minor(version)}.x`;
+        // Version was not found, try again with the closest patch version
+        closestVersion = semver.maxSatisfying(versions, semverRange) as string;
+
+        findVersion(closestVersion);
```
Comment (mshanemc) [inline src/shared/parseReleaseNotes.ts:51]: recursion is probably one of the dangers--you've caught all possible throws but an ♾️ loop would also be really, really bad.

File: src/shared/parseReleaseNotes.ts [hunk 2]
```diff
@@ -6,37 +6,56 @@
...
-      found = false;
-    } else if (found === true) {
-      return token;
+    if (!tokens.length) {
+      if (findClosest === false) {
+        findClosest = true;
+
+        const semverRange = `${semver.major(version)}.${semver.minor(version)}.x`;
+        // Version was not found, try again with the closest patch version
+        closestVersion = semver.maxSatisfying(versions, semverRange) as string;
```
Comment (mshanemc) [inline src/shared/parseReleaseNotes.ts:49]:
```suggestion
        closestVersion = semver.maxSatisfying<string>(versions, semverRange);
```

This lets you pass in the output type rather than asserting it.  It's up to the library to constrain what types it accepts.

File: src/shared/parseReleaseNotes.ts [hunk 3]
```diff
@@ -6,37 +6,56 @@
...
-  const escapeRegExp = (string: string): string => {
-    return string.replace(/[.*+?^${}()|[\]\\]/g, '\\$&'); // $& means the whole matched string
-  };
+  let tokens: marked.Token[];
+
+  const findVersion = (desiredVersion: string): void => {
+    tokens = parsed.filter((token) => {
+      // TODO: Could make header depth (2) a setting in oclif.info.releasenotes
+      if (token.type === 'heading' && token.depth === 2) {
+        const coercedVerion = semver.coerce(token.text).version;
```
Comment (mshanemc) [inline src/shared/parseReleaseNotes.ts:25]:
```suggestion
        const coercedVersion = semver.coerce(token.text).version;
```

File: src/commands/info/releasenotes/display.ts [hunk 4]
```diff
@@ -86,7 +86,7 @@ export default class Display extends SfdxCommand {
         renderer: new TerminalRenderer({ emoji: false }),
       });
 
-      this.ux.log(marked.parse(`# Release notes for '${this.config.bin}':`));
+      tokens.unshift(marked.lexer(`# Release notes for '${this.config.bin}':`)[0]);
```
Comment (mshanemc) [inline src/commands/info/releasenotes/display.ts:89]: any chance marked.lexer is `undefined` and 
1. that [0] is out of bounds?
2. you're unshifting an `undefined` to the front of the tokens?

**Review by mshanemc — CHANGES_REQUESTED** [review]
fix the typo and remove the recursion

File: src/commands/info/releasenotes/display.ts (hunk 4 above)
Comment (iowillhoit) [reply src/commands/info/releasenotes/display.ts:89]: We are calling this method with the exact same thing every time, so it would only break if the `marked.lexer` changed. I will add a test to make sure we always have a zero index, that way we will know early

Comment (iowillhoit) [reply src/commands/info/releasenotes/display.ts:89]: Ah, actually we already have tests that will confirm this. [Here](https://github.com/salesforcecli/plugin-info/pull/45/files#diff-ddc61097337c2e48df39f89511c6113ec1c53efd4f1a3ac9c78e3696e674d123R120) and [here](https://github.com/salesforcecli/plugin-info/blob/main/test/commands/info/releasenotes/display.test.ts#L166)

File: src/shared/parseReleaseNotes.ts [hunk 5]
```diff
@@ -6,36 +6,57 @@
  */
 
 import { marked } from 'marked';
+import * as semver from 'semver';
 
 const parseReleaseNotes = (notes: string, version: string, baseUrl: string): marked.Token[] => {
   let found = false;
+  let findClosest = false;
```
Comment (mshanemc) [inline src/shared/parseReleaseNotes.ts:13]: I think you can delete this var entirely.  

then, line 72 can just ask if `closestVersion` ever got defined

**Review by mshanemc — APPROVED** [review]
yeah, that feels a lot easier to follow for my poor linear brain

File: src/shared/parseReleaseNotes.ts (hunk 5 above)
Comment (iowillhoit) [reply src/shared/parseReleaseNotes.ts:13]: Good call 🎉

**Review by mshanemc — APPROVED** [review]

**Comment by mshanemc** [comment]
QA Notes:
using sfdx info:releasenotes:display 
✅ tested with no version, still ok
✅ tested with known existing version (7.130.0)
✅ checked known invalid version (7.129.1) and saw expected message `# ATTENTION: Version 7.129.1 was not found. Showing notes for closest patch version 7.129.0.`
